test/inductor/test_profiler.py

```diff
@@ -198,6 +201,80 @@ def fn(x, y):
         self.assertTrue(hooks_called["enter"])
         self.assertTrue(hooks_called["exit"])
 
+    @unittest.skipIf(not HAS_TRITON, "requires cuda & triton")
+    def test_pt2_triton_attributes(self):
+        from torch._inductor.codecache import code_hash
+
+        device = "cuda"
+
+        @torchdynamo.optimize("inductor")
+        def fn(a, b, c):
+            x = torch.nn.functional.linear(a, b)
+            x = x + c
+            return x.cos()
+
+        a, b, c = (torch.randn(4, 4, requires_grad=True).to(device) for _ in range(3))
+
+        inputs = [a, b, c]
+        with config.patch(compile_threads=1):
+            fn(*inputs)
+
+        fp = tempfile.NamedTemporaryFile("w+t", suffix=".json", delete=False)
```

Comment:
nit: this temp file is never cleaned up? Why not just use a tmpdir fixture?

Reply:
Yes this was mainly to debug the generated trace file in case of errors, i now added a debug variable that is default False, this will ensure the tempfile is delete.